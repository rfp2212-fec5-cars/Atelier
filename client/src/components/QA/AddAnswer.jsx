import React from 'react';

const AddAnswer = () => {
  return (
    <button type='submit'>Add Answer</button>
  )
}

export default AddAnswer;