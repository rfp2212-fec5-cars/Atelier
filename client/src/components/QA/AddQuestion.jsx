import React from 'react';

const AddQuestion = () => {
  return (
    <button type='submit'>Ask Your Question</button>
  )
};

export default AddQuestion;