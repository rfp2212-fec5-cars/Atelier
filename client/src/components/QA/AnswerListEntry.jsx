import React from 'react';

const AnswerListEntry = ({ answer }) => {
  return (
    <div>
      <h5>A: { answer.body }</h5>
      <label>Helpful? </label>
      <button type='submit'>Yes</button>
      { answer.helpfulness }
      <div>
        <button type='submit'>Report</button>
      </div>
    </div>
  )
};

export default AnswerListEntry;