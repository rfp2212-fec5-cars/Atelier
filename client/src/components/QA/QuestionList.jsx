import React from 'react';

const QuestionList = ({ questionList }) => {
  return (
    <div>
      <h2>Q: </h2>
      { questionList.map((question) => (
        <li>{ question.question_body} </li>
      ))}
    </div>
  )
};

export default QuestionList;