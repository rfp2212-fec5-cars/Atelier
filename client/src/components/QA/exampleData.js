module.exports = {
  product_id: 40390,
  product_name: 'Cleta Skirt',
  questionList: [
    {
      question_id: 329393,
      question_body: 'Beatae autem recusandae nesciunt.',
      question_date: '2021-02-05T00:00:00.000Z',
      asker_name: 'Jermaine.Stehr89',
      question_helpfulness: 27,
      reported: false,
      answers: [{
        answer_id: 3077112,
        body: 'Sapiente natus error.',
        date: '2020-11-12T00:00:00.000Z',
        answerer_name: 'Jamal_Weimann69',
        helpfulness: 19,
        photos: [Array]
      },
      {
        answer_id: 3077110,
        body: 'In quo nisi eum est eaque nihil est.',
        date: '2020-12-20T00:00:00.000Z',
        answerer_name: 'Johathan_Ebert50',
        helpfulness: 9,
        photos: [Array]
      },
      {
        answer_id: 3077109,
        body: 'Aliquam consequatur dolores beatae et occaecati aperiam.',
        date: '2021-06-13T00:00:00.000Z',
        answerer_name: 'Aracely.Schinner',
        helpfulness: 3,
        photos: []
      },
      {
        answer_id: 3077108,
        body: 'Sunt molestiae nobis.',
        date: '2020-08-13T00:00:00.000Z',
        answerer_name: 'Mack.Bailey43',
        helpfulness: 2,
        photos: [Array]
      },
      {
        answer_id: 3077107,
        body: 'Autem quo in omnis voluptatem asperiores perferendis.',
        date: '2021-06-11T00:00:00.000Z',
        answerer_name: 'Benedict15',
        helpfulness: 0,
        photos: []
      }]
    },
    {
      question_id: 329392,
      question_body: 'Earum velit sapiente rerum repellendus.',
      question_date: '2021-06-12T00:00:00.000Z',
      asker_name: 'Marianna_Kunde45',
      question_helpfulness: 25,
      reported: false,
      answers: {}
    },
    {
      question_id: 329394,
      question_body: 'Eum harum cupiditate dicta totam modi nesciunt.',
      question_date: '2020-12-01T00:00:00.000Z',
      asker_name: 'Randal.Wuckert',
      question_helpfulness: 24,
      reported: false,
      answers: [{
        answer_id: 3077114,
        body: 'Quasi reiciendis aut.',
        date: '2021-02-08T00:00:00.000Z',
        answerer_name: 'Creola83',
        helpfulness: 14,
        photos: [Array]
      },
      {
        answer_id: 3077117,
        body: 'Voluptatem doloremque nostrum et consequatur aspernatur fuga tempora sequi.',
        date: '2021-05-17T00:00:00.000Z',
        answerer_name: 'Ivah.Heaney40',
        helpfulness: 9,
        photos: []
      },
      {
        answer_id: 3077113,
        body: 'Laboriosam dolorem sapiente nobis esse enim qui.',
        date: '2021-01-26T00:00:00.000Z',
        answerer_name: 'Trudie_Davis',
        helpfulness: 8,
        photos: [Array]
      },
      {
        answer_id: 3077115,
        body: 'Voluptatibus voluptatum a aut aut sapiente.',
        date: '2021-04-21T00:00:00.000Z',
        answerer_name: 'Carmela.Hayes92',
        helpfulness: 8,
        photos: []
      },
      {
        answer_id: 3077118,
        body: 'Rerum vel quo quasi fugit ut explicabo ea doloribus accusamus.',
        date: '2021-03-06T00:00:00.000Z',
        answerer_name: 'Jaeden86',
        helpfulness: 2,
        photos: [Array]
      }]
    },
    {
      question_id: 329395,
      question_body: 'Commodi laborum aspernatur et fuga quae aperiam quae.',
      question_date: '2020-11-13T00:00:00.000Z',
      asker_name: 'Claudia.Gibson6',
      question_helpfulness: 19,
      reported: false,
      answers: [{
        answer_id: 3077121,
        body: 'Porro ut voluptate quia at sed porro maiores.',
        date: '2021-01-28T00:00:00.000Z',
        answerer_name: 'Filomena.Daugherty66',
        helpfulness: 17,
        photos: [Array]
      },
      {
        answer_id: 3077120,
        body: 'Qui accusamus consequuntur eveniet doloremque delectus eveniet ab.',
        date: '2020-12-04T00:00:00.000Z',
        answerer_name: 'Tressie26',
        helpfulness: 9,
        photos: [Array]
      },
      {
        answer_id: 3077122,
        body: 'Quia nobis aut vitae veniam facere hic dolorem qui.',
        date: '2021-05-28T00:00:00.000Z',
        answerer_name: 'Jensen_Sauer',
        helpfulness: 2,
        photos: []
      }]
    },
    {
      question_id: 329398,
      question_body: 'Iste aut accusamus non nostrum ea.',
      question_date: '2021-01-03T00:00:00.000Z',
      asker_name: 'Clarabelle.Swaniawski',
      question_helpfulness: 15,
      reported: false,
      answers: [Object]
    },
    {
      question_id: 329397,
      question_body: 'Repudiandae voluptatem accusantium tempora laudantium non doloribus.',
      question_date: '2021-03-26T00:00:00.000Z',
      asker_name: 'June_Bednar',
      question_helpfulness: 12,
      reported: false,
      answers: [Object]
    },
    {
      question_id: 329396,
      question_body: 'Qui quis ut sit molestiae blanditiis voluptatum odio et.',
      question_date: '2021-07-10T00:00:00.000Z',
      asker_name: 'Nathan.Mayert',
      question_helpfulness: 8,
      reported: false,
      answers: [Object]
    }
  ],

  answerList: [

  ]
}