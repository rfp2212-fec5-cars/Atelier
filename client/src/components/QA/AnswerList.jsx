import React from 'react';
import AnswerListEntry from './AnswerListEntry.jsx';

const AnswerList = ({ answers }) => {
  return (
    <div>
      <h2>Answer List</h2>
      { answers.map(answer => (
        <AnswerListEntry key={ answer.answer_id } answer={ answer }/>
      ))}
      <button type='submit'>See More Answers</button>
    </div>
  );
};

export default AnswerList;