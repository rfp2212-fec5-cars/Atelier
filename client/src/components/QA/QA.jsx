import React, { useState, useEffect } from 'react';
import axios from 'axios';
import Search from './Search.jsx';
import QuestionList from './Questions/QuestionList.jsx';
import MoreQuestions from './Questions/MoreQuestions.jsx';
import AddQuestion from './Questions/AddQuestion.jsx';

const QA = ({ productId }) => {
  const [questionList, setQuestionList] = useState([]);
  const [displayedQuestions, setDisplayedQuestions] = useState([]);
  const [updateQuestions, setUpdateQuestions] = useState(false);

  useEffect(() => {
    var options = {
      url: '/qa/questions',
      params: {
        product_id: productId,
        page: 1,
        count: 100
      }
    };

    axios(options)
      .then(({ data }) => {
        var sortedQuestions = data.results.sort((a, b) => {
          return b.question_helpfulness - a.question_helpfulness;
        });

        setQuestionList(sortedQuestions);
        setDisplayedQuestions(sortedQuestions.slice(0, 4));
      })
      .catch((err) => {
        console.log(err);
      })

  }, [updateQuestions]);

  return (
    <div>
      <h1>Questions & Answers</h1>
      <Search />
      <QuestionList questionList={ displayedQuestions } product_id={ productId }/>
      <MoreQuestions questionList={ questionList } setDisplayedQuestions={ setDisplayedQuestions } displayedQuestions={ displayedQuestions }/>
      <AddQuestion product_id={ productId } updateQuestions={ updateQuestions } setUpdateQuestions={ setUpdateQuestions }/>
    </div>
  )
};

export default QA;