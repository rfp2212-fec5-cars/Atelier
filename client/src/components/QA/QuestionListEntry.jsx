import React, { useState, useEffect } from 'react';
import axios from 'axios';
import AnswerList from './AnswerList.jsx';
import AddAnswer from './AddAnswer.jsx';

const QuestionListEntry = ({ question }) => {
  const [answerList, setAnswerList] = useState([]);

  useEffect(() => {
    var options = {
      url: `/qa/questions/${question.question_id}/answers`,
      params: {
        page: 1,
        count: 100
      }
    }
    axios(options)
      .then(({ data }) => setAnswerList(data.results))
      .catch((err) => {
        console.log('Failed to get answer list', err);
      })
  }, [])

  return (
    <div>
      <h4>Q: { question.question_body }</h4 >
      <div>
        <label>Helpful? </label>
        <button type='submit'>Yes</button>
        { question.question_helpfulness }
        <div>
          <AddAnswer />
        </div>
      </div>
      <AnswerList answers={ answerList }/>
      <hr></hr>
    </div>
  );
};

export default QuestionListEntry;