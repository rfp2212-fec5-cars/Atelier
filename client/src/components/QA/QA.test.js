import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import React from 'react';
import ReactDOM from 'react-dom';
import QA from './QA.jsx';

afterEach(cleanup);

test('should not crash on load', () => {
  render(<QA />);
})