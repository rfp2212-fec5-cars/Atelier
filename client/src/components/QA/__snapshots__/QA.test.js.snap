// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the search component in the Q&A section 1`] = `
<form>
  <input
    onChange={[Function]}
    placeholder="Have a Question? Search for answers..."
    type="text"
  />
</form>
`;
